tests/php/test-class-amp-accessibility-sanitizer.php

```diff
@@ -68,4 +104,83 @@ public function test_sanitize( $source, $expected ) {
 
 		$this->assertEqualMarkup( $expected, $actual );
 	}
+
+	/**
+	 * Data provider for $this->test_add_skip_link()
+	 *
+	 * @return string[][]
+	 */
+	public function get_skip_link_test_data() {
+
+		$style_element = '<style id="skip-link-styles">.skip-link.screen-reader-text {
+				border: 0;
+				clip: rect(1px,1px,1px,1px);
+				clip-path: inset(50%);
+				height: 1px;
+				margin: -1px;
+				overflow: hidden;
+				padding: 0;
+				position: absolute !important;
+				width: 1px;
+				word-wrap: normal !important;
+			}
+
+			.skip-link.screen-reader-text:focus {
+				background-color: #eee;
+				clip: auto !important;
+				clip-path: none;
+				color: #444;
+				display: block;
+				font-size: 1em;
+				height: auto;
+				left: 5px;
+				line-height: normal;
+				padding: 15px 23px 14px;
+				text-decoration: none;
+				top: 5px;
+				width: auto;
+				z-index: 100000;
+			}</style>';
+
+		return [
+			'with_id'              => [
+				'source'   => '<html><body><main id="main-container">Hello World!</main></body></html>',
+				'expected' => $style_element . '<a class="skip-link screen-reader-text" href="#main-container">Skip to content</a><main id="main-container">Hello World!</main>',
+			],
+			'without_id'           => [
+				'source'   => '<html><body><main>Hello World!</main></body></html>',
+				'expected' => $style_element . '<a class="skip-link screen-reader-text" href="#wp--skip-link--target">Skip to content</a><main>Hello World!</main>',
+			],
+			'without_main_element' => [
+				'source'   => '<html><body><div id="main-container">Hello World!</div></body></html>',
+				'expected' => '<div id="main-container">Hello World!</div>',
+			],
+		];
+
+	}
+
+	/**
+	 * @dataProvider get_skip_link_test_data
+	 *
+	 * @covers ::add_skip_link()
+	 *
+	 * @param string $source   Source HTML.
+	 * @param string $expected Expected target HTML.
+	 */
+	public function test_add_skip_link( $source, $expected ) {
+
+		global $_wp_current_template_content, $_wp_theme_features;
+
+		$_wp_current_template_content          = true;
+		$_wp_theme_features['block-templates'] = true;
```

Comment:
Should we also add a test case for when a theme or a template is not block-based?